unit/solvers/smt2_incremental/smt_sorts.cpp

```diff
@@ -30,3 +30,50 @@ TEST_CASE(
 {
   REQUIRE(smt_bit_vector_sortt{32}.bit_width() == 32);
 }
+
+TEST_CASE("Visiting smt_bool_sortt.", "[core][smt2_incremental]")
+{
+  class : public smt_sort_const_downcast_visitort
```

Comment:
⛏️ Given that the visitor is nearly the same in both of the following test cases, does it make sense for it to be pulled out somewhere?

Reply:
Yes. I usually err on the side of more duplication in unit tests, in order to make it more obvious why the properties the test is testing should hold. The duplication isn't too bad for the sorts at the moment as there are only a couple of cases. But it will be much worse for the terms and commands and it will be worse if we expand this with more sorts. Therefore I'll come up with a way to factor it out for the terms and commands and then back port the solution to the sorts.